libraries/AC_Avoidance/AC_Avoidance_Logging.cpp

```diff
@@ -3,6 +3,7 @@
 #include "AP_OABendyRuler.h"
 #include <AP_Logger/AP_Logger.h>
 
+#if AP_OAPATHPLANNER_ENABLED
```

Comment:
```suggestion
#if AP_OAPATHPLANNER_ENABLED  && HAL_LOGGING_ENABLED
```

... elsewhere too

Reply:
Ummm... Can we do it in another PR? If we do this here, then we need to compile out the declarations of these methods from `.h` files and the places we are calling these methods based on `#if HAL_LOGGING_ENABLED`. Shouldn't all that go into a separate PR?